pxd.c

```diff
@@ -406,9 +407,57 @@ static void pxd_req_misc(struct fuse_req *req, uint32_t size, uint64_t off,
 #endif
 }
 
-static void pxd_read_request(struct fuse_req *req, uint32_t size, uint64_t off,
+/*
+ * when block device is registered in non blk mq mode, device limits are not
+ * honoured. Handle it appropriately.
+ */
+static
+int pxd_handle_device_limits(struct fuse_req *req, uint32_t *size, uint64_t *off,
```

Comment:
Isn't pxd_fastpath_adjust_limits() supposed to take care of the q limits?

Reply:
Originally we had bio too large error message (refer: https://access.redhat.com/solutions/3613951), that was fixed by introducing these to adjust limits based on actual targets. 
In this case, even though limits were set, under some scenarios, block device drivers requests were over the limits for read/write and even discard. This caused px-storage to assert at various places. Maybe the issue is kernel version dependent, but the fix is generic.